### ISSUE TITLE
Question: Nested Placeholders

### ISSUE THREAD

**Issue by Mike-E-angelo** [axuno/SmartFormat#354]
Thank you for creating such a great library 🙏

I am attempting to use nested placeholders as described here:
https://github.com/axuno/SmartFormat/issues/10#issuecomment-21823301

Here is my template:

```
A request was made to send sign-in information associated with this email address.

Here is the known information:

{Authentications:
	{Provider}:
	Identity: {Identity}
	Display Name: {DisplayName}
	Identifier: {Identifier}	

}

If you made this request, no further action is required at this time.
```

However, when I attempt to run this, I get the following error:

```
SmartFormat.Core.Formatting.FormattingException: Error parsing format string: No source extension could handle the selector named "Provider" at <char>
```

Here are the classes I am using, passing in a `SignInInformation`:

```
public sealed record AuthenticationView
	(string Provider, string Identity, string DisplayName, string Description, string? Identifier);

public readonly record struct SignInInformation(string Address, AuthenticationView[] Authentications);
```

Is there an additional configuration that is missing, perhaps?

Thank you for any assistance you can provide.

**Comment by Mike-E-angelo** [comment]
https://stackoverflow.com/a/19860111

**Comment by axunonb** [comment]
Happy to see you found the remedy yourself. Another example for handling lists:
```C#
[Test, Description("Format a list of lists")]
public void List_Of_Lists_With_Element_Format()
{
    var data = new List<int[]> {
        new[] { 1, 2, 3 },
        new[] { 4, 5, 6 },
        new[] { 7, 8, 9 }
    };

    var formatter = new SmartFormatter()
        .AddExtensions(new ListFormatter(), new DefaultSource())
        .AddExtensions(new DefaultFormatter());

    var expected = "001, 002, 003\n" + "004, 005, 006\n" + "007, 008, 009";
    var result = formatter.Format("{0:list:{:list:{:000}|, |, }|\n|\n}", data);
    //                                   |       |        | |       |      |
    //                                   |       |  element format  |      |
    //                                   |       |___ inner list ___|      |
    //                                   |___________ outer list __________|
    Assert.That(result, Is.EqualTo(expected));
}
```

**Comment by Mike-E-angelo** [comment]
Thank you very much for that @axunonb.  I did see that in the documentation but was having trouble in my case as the sample uses numbers when I am using objects.  I landed on the right answer eventually. :)

I would like to thank you again for this project.  I use it to format .md files stored in Resx as templates and then run objects through them at runtime to generate emails to send to users.  It works really well. 👍

FWIW I have this project listed in my application's acknowledgements, along with half of NuGet it seems 😅
https://starbeam.one/about/acknowledgements
